File: src/lib/coords.ts
import { BUILDING } from "@/data/dimensions";

/**
 * Plan space: +X east, LDK at small X (west), genkan east of LDK.
 *
 * Three.js cameras looking north put world +X on the *left* of the screen
 * (opposite of standing outdoors looking north). To keep both top-down and
 * first-person matching the floor plan, we render the house mirrored in X:
 *
 *   worldX = BUILDING.width - planX
 *
 * After mirroring, looking north: left = LDK, right = genkan. ✓
 * Top-down north-up: left = LDK, right = genkan. ✓
 *
 * Store wall data in plan space; convert only when placing free-world
 * entities (player, camera target helpers) outside the mirrored group.
 */

/** Plan X → world X (after house mirror group). */
export function planToWorldX(planX: number): number {
  return BUILDING.width - planX;
}

/** World X → plan X. */
export function worldToPlanX(worldX: number): number {
  return BUILDING.width - worldX;
}

File: shared/src/commonMain/kotlin/com/neojou/japanesehouse3d/domain/Coords.kt
package com.neojou.japanesehouse3d.domain

/**
 * Plan space: +X east, +Z north, +Y up; origin SW of LDK.
 *
 * npm display mirrors house in X (`planToWorldX = width - planX`) so north view
 * matches PDF (LDK left). K2 renderer uses **plan space** for camera + mesh;
 * apply [planToWorldX] only when interop with npm world or future mirror group.
 */
fun planToWorldX(planX: Double): Double = Building.width - planX

fun worldToPlanX(worldX: Double): Double = Building.width - worldX
